src/main/java/com/marklogic/mgmt/resource/security/CredentialsManager.java

```diff
@@ -46,30 +38,23 @@ protected String getIdFieldName() {
 
 	@Override
 	protected String getResourceId(String payload) {
-		if (payloadParser.isJsonPayload(payload)) {
-			return payloadParser.getPayloadFieldValue(payload, getIdFieldName());
-		} else {
-			if (payloadParser.getPayloadFieldValue(payload, "aws", false) != null) {
-				return "aws";
-			} else if (payloadParser.getPayloadFieldValue(payload, "azure", false) != null) {
-				return "azure";
-			} else {
-				return null;
-			}
-		}
+		return getCredentialsType(payload);
 	}
 
 	@Override
-	protected SaveReceipt createNewResource(String payload, String resourceId) {
-		String label = getResourceName();
-		if (logger.isInfoEnabled()) {
-			logger.info(format("Creating %s: %s", label, resourceId));
-		}
-		String path = getCreateResourcePath(payload);
-		ResponseEntity<String> response = putPayload(getManageClient(), path, payload);
-		if (logger.isInfoEnabled()) {
-			logger.info(format("Created %s: %s", label, resourceId));
+	public DeleteReceipt delete(String payload, String... resourceUrlParams) {
+		final String type = getCredentialsType(payload);
+		final String path = "/manage/v2/credentials/properties?type=" + getCredentialsType(payload);
+		// The DELETE endpoint - https://docs.marklogic.com/REST/DELETE/manage/v2/credentials/properties - seems to
```

Comment:
I frequently include comments like this whenever something is being done because of something kind of wacky in the Manage API.